src/pages/ExamsPage/components/AddAllowanceModal.jsx

```diff
@@ -0,0 +1,239 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ModalDialog,
+  Form,
+  ActionRow,
+  StatefulButton,
+} from '@openedx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import * as constants from '../../../data/constants';
+import { useExamsData, useCreateAllowance, useButtonStateFromRequestStatus } from '../hooks';
+import messages from '../messages';
+
+const AddAllowanceModal = ({ isOpen, close }) => {
+  const { examsList } = useExamsData();
+  const proctoredExams = [];
+  const timedExams = [];
+
+  for (let i = 0; i < examsList.length; i++) { // maybe move to utils
+    const exam = examsList[i];
+    if (exam.examType === 'proctored') {
+      proctoredExams.push(exam);
+    } else {
+      timedExams.push(exam);
+    }
+  }
```

Comment:
Just a suggestion. If hook data needs to be processed to be used in a component, maybe a cleaner approach could be to make it it's own hook. It would be much easier to test as well, like:

```javascript
export const useFilteredExamsData => () => {
  const { examsList } = useExamsData();

  const proctoredExams = examsList.filter(e => e.examType === 'proctored')
  const timedExams = examsList.filter(e => e.examType !== 'proctored')
  
  return { proctoredExams, timedExams }
}
```

and then just call it in the component as:
```javascript
const { proctoredExams, timedExams } = useFilteredExamsData() 
```

No need to change it. Just a suggestion.

Reply:
Thank you! I like how this is cleaner, so I will update that.